dev_docs/key_concepts/anatomy_of_a_plugin.mdx

```diff
@@ -39,27 +39,41 @@ plugins/
+  "version": "1.0.0",
+  "kibanaVersion": "7.14.0",
   "server": true,
   "ui": true,
-  "owner": { [1]
+  "owner": {
     "name": "App Services",
     "githubTeam": "kibana-app-services"
   },
-  "description": "This plugin extends Kibana by doing xyz, and allows other plugins to extend Kibana by offering abc functionality. It also exposes some helper utilities that do efg", [2]
-  "requiredPlugins": ["data"], [3]
-  "optionalPlugins": ["alerting"] [4]
-  "requiredBundles": ["anotherPlugin"] [5]
+  "description": "A description about this plugin!",
+  "requiredPlugins": ["data"],
+  "optionalPlugins": ["alerting"]
+  "requiredBundles": ["anotherPlugin"]
 }
 ```
 
-[1], [2]: Every internal plugin should fill in the owner and description properties.
+`id` - [Required] The id of your plugin can be anything, though it should be fairly unique, as every plugin in an installation needs to be unique. It must be snakeCase.
+
+`version` - [Required] Note the version of your plugin. Kibana doesn't use this field, but it is required.
+
+`kibanaVersion` - [Optional] If your plugin is only compatible with a specific version of Kibana, put it here. Note that internal, first-party plugins shouldn't use this, as they are expected to be compatible with every Kibana version.
+
+`server` - [Optional] If your plugin contains server-side code, this must be true.
+
+`ui` - [Optional] If your plugin contains client-side code, this must be true.
+
+`owner` - [Required] Help users of your plugin know who manages this plugin and how to get in touch. This is required for internal plugins.
+
+`description` - [Required] Give your plugin a description to help other developers understand what it does. This is required for internal plugins.
 
-[3], [4]: Any plugin that you have a dependency on should be listed in `requiredPlugins` or `optionalPlugins`. Doing this will ensure that you have access to that plugin's start and setup contract inside your own plugin's start and setup lifecycle methods. If a plugin you optionally depend on is not installed or disabled, it will be undefined if you try to access it. If a plugin you require is not installed or disabled, kibana will fail to build.
+`requiredPlugins` - [Optional] If your plugin requires any other plugins to work, you must list them here by id. If any of the required plugins are disabled or not installed, then your plugin will be disabled.
 
-[5]: Don't worry too much about getting 5 right. The build optimizer will complain if any of these values are incorrect.
+`optionalPlugins` - [Optional] If your plugin has an optional dependency on other plugins, you must list them here by id. If any of the optional plugins are disabled or not installed, your plugin will still load, however that plugin's services will be undefined in the second parameter of the setup and start functions.
 
+`requiredBundles` - Don't worry about getting this right. The build optimizer will complain if any of these values are incorrect.
```

Comment:
`configPath` - every plugin might allow Kibana users to adjust configuration via `kibana.yml` file. If your plugin needs to read config from `kibana.yml , you should declare what property name it should have access to.